### PR TITLE
Updates to package.py

### PR DESCRIPTION
**Pull request by mauneyc-LANL**
## PR Summary

- Add variants for `spiner`
- Refine some depends_on to not grab unnecessary packages
- Make sure CMake args are correct

@dholladay00 @jhp-lanl @jonahm-LANL @rbberger 
Comments/merges/changes. We can discuss more during scheduled meetings.

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by using the `make format` command after configuring with `cmake`.
- [ ] Document any new features, update documentation for changes made.
- [ ] Make sure the copyright notice on any files you modified is up to date.
- [ ] After creating a pull request, note it in the CHANGELOG.md file
- [ ] If preparing for a new release, update the version in cmake.

### CODE REVIEW

**Comment by Yurlungur** [comment]
This all seems reasonable to me. Thanks for attacking this so quickly

**Comment by mauneyc-LANL** [comment]
> This all seems reasonable to me. Thanks for attacking this so quickly

@Yurlungur I'd like to have a discussion/input on this before moving forward. Since we have a meet scheduled for day I think that'd be a good place to hammer this into shape.

**Comment by Yurlungur** [comment]
Yep sounds good

**Comment by jhp-lanl** [comment]
> This all seems reasonable to me. Thanks for attacking this so quickly

To echo Chris, yes please don't merge this yet. I plan to provide some input too

**Comment by jhp-lanl** [comment]
Another issue I just found when trying to link with EAP: Cuda gets turned on by default in the CMake when Kokkos is on. I _think_ there isn't logic to allow building with Kokkos but without Cuda in the package.py right now. Please correct me if I'm wrong though

**Comment by Yurlungur** [comment]
You can do `-DSINGULARITY_ENABLE_KOKKOS=ON -DSINGULARITY_ENABLE_CUDA=OFF` to override the cmake dependent option default settings. But I think the logic here probably should be reversed.

**Comment by jhp-lanl** [comment]
> You can do `-DSINGULARITY_ENABLE_KOKKOS=ON -DSINGULARITY_ENABLE_CUDA=OFF` to override the cmake dependent option default settings. But I think the logic here probably should be reversed.

I'm fine with either to be honest as long as the hooks are there to ensure we can build without Cuda both through CMake (exists) and through spack. I think that logic doesn't exist in the `package.py` yet.

File: spack-repo/packages/singularity-eos/package.py
Comment (jhp-lanl) [inline spack-repo/packages/singularity-eos/package.py:80]: Do we want to specifically avoid 1.5.1 here since it has the inline issue? Something like
```suggestion
    depends_on("ports-of-call@1.4.2,1.5.2:", when="@:1.7.0")
```

**Review by jhp-lanl — APPROVED** [review]
One last request:

Can you update the build documentation for a few things:

1. Change "Overview" to something about building. "Overview" becomes the page title and it's a bit too broad.
2. Change spiner to an optional dependency and add something like "Optional; used for EOS that rely on databox or spiner"
3. Move `SINGULARITY_BUILD_CLOSURE` to the standard options rather than dependent options
```
``SINGULARITY_BUILD_CLOSURE``           ON       Build the mixed cell closure models
```
4. Reflect the fact that `SINGULARITY_USE_KOKKOSKERNELS` depends on `SINGULARITY_BUILD_CLOSURE` and not the other way around

Approving now so I don't forget to approve later

**Review by Yurlungur — APPROVED** [review]
Approved pending @jhp-lanl 's additional requests

**Comment by mauneyc-LANL** [comment]
@jhp-lanl @jonahm-LANL Docs updated

**Comment by jhp-lanl** [comment]
@dholladay00 or @rbberger if you have any comments can you make them by the end of today? Otherwise if not I think we can merge this.

**Review by rbberger — APPROVED** [review]
LGTM

**Comment by jhp-lanl** [comment]
I'll go ahead and merge this in. If there are any changes you want @dholladay00 we can open up some issues

**Pull request merged**
